src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Telemetry/MetricConstants.cs

```diff
@@ -0,0 +1,167 @@
+
+            public const string Timeouts = StandardsPrefix + "timeouts";
+
+            public const string CreationTime = StandardsPrefix + "create_time";
+
+            public const string WaitTime = StandardsPrefix + "wait_time";
+
+            public const string UsageTime = StandardsPrefix + "use_time";
+
+            public const string HardUsage = LibrarySpecificPrefix + "hard.usage";
+
+            public const string Connects = LibrarySpecificPrefix + "connects";
+
+            public const string Disconnects = LibrarySpecificPrefix + "disconnects";
+
+            public const string ServerCommandTime = LibrarySpecificPrefix + "server_command_time";
+
+            public const string NetworkWaitTime = LibrarySpecificPrefix + "network_wait_time";
+
+            public const string ByteIo = LibrarySpecificPrefix + "io";
+
+            public const string BuffersIo = LibrarySpecificPrefix + "buffers";
+
+            public const string HardTotal = LibrarySpecificPrefix + "hard.total";
+
+            public const string SoftTotal = LibrarySpecificPrefix + "soft.total";
+        }
+
+        public static class ConnectionPools
+        {
+            private const string LibrarySpecificPrefix = MetricNames.LibrarySpecificPrefix + "connection_pools.";
+
+            public const string Usage = LibrarySpecificPrefix + "usage";
+        }
+
+        public static class ConnectionPoolGroups
+        {
+            private const string LibrarySpecificPrefix = MetricNames.LibrarySpecificPrefix + "connection_pool_groups.";
+
+            public const string Usage = LibrarySpecificPrefix + "usage";
+        }
+
+        public static class Commands
+        {
+            private const string LibrarySpecificPrefix = MetricNames.LibrarySpecificPrefix + "commands.";
+
+            public const string Failed = LibrarySpecificPrefix + ".failed";
+
+            public const string UsageTime = LibrarySpecificPrefix + "use_time";
+
+            public const string PreparedRatio = LibrarySpecificPrefix + "prepared_ratio";
+
+            public const string Total = LibrarySpecificPrefix + "total";
+        }
+
+        public static class Transactions
+        {
+            private const string LibrarySpecificPrefix = MetricNames.LibrarySpecificPrefix + "transactions.";
+
+            public const string Committed = LibrarySpecificPrefix + ".committed";
+
+            public const string Active = LibrarySpecificPrefix + ".active";
+
+            public const string RolledBack = LibrarySpecificPrefix + ".rolled_back";
+
+            public const string Total = LibrarySpecificPrefix + "total";
+
+            public const string CommitTime = LibrarySpecificPrefix + "commit_time";
+
+            public const string RollbackTime = LibrarySpecificPrefix + "rollback_time";
+
+            public const string ActiveTime = LibrarySpecificPrefix + "active_time";
+        }
+    }
+
+    internal static class MetricAttributes
```

Comment:
`MetricsAttributesNames`? The Attributes suffix implies that it contains attributes to me.

Reply:
Yes, thanks. I spent a few minutes dithering between describing them as attributes or tags, before deciding to use tag-based terminology for consistency. I'll rename it and its counterpart to `MetricTagNames` and `MetricTagValues` respectively.